stoobly_agent/app/cli/scaffold/hosts_file_reader.py

```diff
@@ -0,0 +1,67 @@
+import pdb
+import platform
+from dataclasses import dataclass
+from typing import Union
+from urllib.parse import urlparse
+
+
+class HostsFileReader():
+
+  @dataclass
+  class IpAddressToHostnames:
+    ip_address: str
+    hostnames: list[str]
+
+  def __get_hosts_file_path(self) -> str:
+    system = platform.system()
+    if system == 'Linux' or system == 'Darwin':
+      return '/etc/hosts'
+    else:
+      print(f"Unsupported system: {system}, for hosts file validation, skipping")
+
+    return ''
+
+  # Split IP address and hostnames. Don't include inline comments
+  def __split_hosts_line(self, line: str) -> list[str]:
+    ip_addr_hosts_split = line.split('#')[0].split()
+    return ip_addr_hosts_split
+
+
+  # Parses hosts file and returns a mapping of IP address to hostnames in a list.
+  def get_hosts(self) -> list[IpAddressToHostnames]:
+    hosts_file_path = self.__get_hosts_file_path()
+
+    if not hosts_file_path:
+      return []
+
+    with open(hosts_file_path, 'r') as f:
+      hostlines = f.readlines()
+
+    # Skip comments and empty lines
+    hostlines = [line.strip() for line in hostlines
+                 if not line.startswith('#') and line.strip() != '']
+
+    hosts = []
+    for line in hostlines:
+      ip_addr_hosts_split = self.__split_hosts_line(line)
+      ip_address = ip_addr_hosts_split[0]
+      hostnames = ip_addr_hosts_split[1:]
+      ipAddressToHostnames = self.IpAddressToHostnames(ip_address, hostnames)
+
+      hosts.append(ipAddressToHostnames)
+
+    return hosts
+
+  def find_host(self, url) -> Union[IpAddressToHostnames, None]:
```

Comment:
What happens if we pass in hostname instead of url? Seeing `urlparse` parse hostnames as `path` instead of `netloc`

Reply:
`url` got passed in because `requests.get()` requires a schema. Should we pass in hostname for this new functionality?

Reply:
Seems like `netloc` is the correct one if we stick with passing down only `url`

```
>>> urlparse('google.com')
ParseResult(scheme='', netloc='', path='google.com', params='', query='', fragment='')

>>> urlparse('http://google.com')
ParseResult(scheme='http', netloc='google.com', path='', params='', query='', fragment='')
```

Reply:
We wouldn't ever have a URL for this use case since `/etc/hosts` doesn't support them